diff -Naur make-4.4/src/config.h.in make-4.4.tpg/src/config.h.in
--- make-4.4/src/config.h.in	2022-10-31 06:23:41.000000000 +0000
+++ make-4.4.tpg/src/config.h.in	2022-11-02 11:53:44.787171326 +0000
@@ -888,7 +888,7 @@
    _GL_ATTRIBUTE_DEALLOC_FREE declares that the function returns pointers that
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
-#if _GL_GNUC_PREREQ (11, 0)
+#if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 # define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 #else
 # define _GL_ATTRIBUTE_DEALLOC(f, i)
