diff -urN make-3.82/config/config.sub make-3.82-aarch64/config/config.sub
--- make-3.82/config/config.sub	2010-07-27 23:42:11.000000000 -0600
+++ make-3.82-aarch64/config/config.sub	2012-10-08 16:37:34.782050976 -0600
@@ -247,6 +247,7 @@
 	# Some are omitted here because they have special meanings below.
 	1750a | 580 \
 	| a29k \
+        | aarch64 | aarch64_be \
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
@@ -338,6 +339,7 @@
 	# Recognize the basic CPU types with company name.
 	580-* \
 	| a29k-* \
+        | aarch64-* | aarch64_be-* \
 	| alpha-* | alphaev[4-8]-* | alphaev56-* | alphaev6[78]-* \
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
