--- make-4.2/Makefile.am.0002~	2016-05-22 21:35:46.676153083 +0200
+++ make-4.2/Makefile.am	2016-05-22 21:37:35.515873977 +0200
@@ -43,7 +43,7 @@ else
 endif
 
 make_SOURCES =	ar.c arscan.c commands.c default.c dir.c expand.c file.c \
-		function.c getopt.c getopt1.c guile.c implicit.c job.c load.c \
+		function.c guile.c implicit.c job.c load.c \
 		loadapi.c main.c misc.c $(ossrc) output.c read.c remake.c \
 		rule.c signame.c strcache.c variable.c version.c vpath.c \
 		hash.c $(remote)
@@ -51,9 +51,9 @@ make_SOURCES =	ar.c arscan.c commands.c
 EXTRA_make_SOURCES = vmsjobs.c remote-stub.c remote-cstms.c
 
 noinst_HEADERS = commands.h dep.h filedef.h job.h makeint.h rule.h variable.h \
-		debug.h getopt.h gettext.h hash.h output.h os.h
+		debug.h hash.h output.h os.h
 
-make_LDADD =	@LIBOBJS@ @ALLOCA@ $(GLOBLIB) @GETLOADAVG_LIBS@ @LIBINTL@ \
+make_LDADD =	@LIBOBJS@ @ALLOCA@ $(GLOBLIB) @GETLOADAVG_LIBS@ \
 		$(GUILE_LIBS)
 # Only process if target is MS-Windows
 if WINDOWSENV
