--- make-4.1/Makefile.am.omv~	2015-05-27 12:45:28.366784323 +0200
+++ make-4.1/Makefile.am	2015-05-27 12:45:53.523994918 +0200
@@ -40,7 +40,7 @@ else
 endif
 
 make_SOURCES =	ar.c arscan.c commands.c default.c dir.c expand.c file.c \
-		function.c getopt.c getopt1.c guile.c implicit.c job.c load.c \
+		function.c guile.c implicit.c job.c load.c \
 		loadapi.c main.c misc.c output.c read.c remake.c rule.c \
 		signame.c strcache.c variable.c version.c vpath.c hash.c \
 		$(remote)
@@ -48,9 +48,9 @@ make_SOURCES =	ar.c arscan.c commands.c
 EXTRA_make_SOURCES = vmsjobs.c remote-stub.c remote-cstms.c
 
 noinst_HEADERS = commands.h dep.h filedef.h job.h makeint.h rule.h variable.h \
-		debug.h getopt.h gettext.h hash.h output.h
+		debug.h hash.h output.h
 
-make_LDADD =	@LIBOBJS@ @ALLOCA@ $(GLOBLIB) @GETLOADAVG_LIBS@ @LIBINTL@ \
+make_LDADD =	@LIBOBJS@ @ALLOCA@ $(GLOBLIB) @GETLOADAVG_LIBS@ \
 		$(GUILE_LIBS)
 # Only process if target is MS-Windows
 if WINDOWSENV
