diff -up make-4.4.1/src/config.h.in.2~ make-4.4.1/src/config.h.in
--- make-4.4.1/src/config.h.in.2~	2023-02-27 00:30:25.210872785 +0100
+++ make-4.4.1/src/config.h.in	2023-02-27 00:30:55.122018077 +0100
@@ -910,7 +910,7 @@
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if _GL_GNUC_PREREQ (11, 0)
+# if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
