diff -up make-4.3/src/makeint.h.omv~ make-4.3/src/makeint.h
--- make-4.3/src/makeint.h.omv~	2020-03-20 11:37:33.124159732 +0100
+++ make-4.3/src/makeint.h	2020-03-20 11:37:46.215165439 +0100
@@ -566,6 +566,8 @@ void print_dir_data_base (void);
 void dir_setup_glob (glob_t *);
 void hash_init_directories (void);
 
+FILE_TIMESTAMP name_mtime (const char *name);
+
 void define_default_variables (void);
 void undefine_default_variables (void);
 void set_default_suffixes (void);
diff -up make-4.3/src/read.c.omv~ make-4.3/src/read.c
--- make-4.3/src/read.c.omv~	2020-03-20 11:37:49.815166981 +0100
+++ make-4.3/src/read.c	2020-03-20 11:38:38.925186923 +0100
@@ -418,6 +418,11 @@ eval_makefile (const char *filename, uns
   /* Success; clear errno.  */
   deps->error = 0;
 
+  /* Managed to read the file. Update mtime, if needed.
+     https://savannah.gnu.org/bugs/?57676 */
+  if (deps->file->last_mtime == NONEXISTENT_MTIME)
+    deps->file->last_mtime = name_mtime(deps->file->name);
+
   /* Avoid leaking the makefile to children.  */
   fd_noinherit (fileno (ebuf.fp));
 
diff -up make-4.3/src/remake.c.omv~ make-4.3/src/remake.c
--- make-4.3/src/remake.c.omv~	2020-03-20 11:38:44.225188957 +0100
+++ make-4.3/src/remake.c	2020-03-20 11:38:52.946192254 +0100
@@ -67,7 +67,6 @@ static enum update_status check_dep (str
                                      FILE_TIMESTAMP this_mtime, int *must_make);
 static enum update_status touch_file (struct file *file);
 static void remake_file (struct file *file);
-static FILE_TIMESTAMP name_mtime (const char *name);
 static const char *library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr);
 
 
@@ -1462,7 +1461,7 @@ f_mtime (struct file *file, int search)
    This causes one duplicate stat() when -L is being used, but the code is
    much cleaner.  */
 
-static FILE_TIMESTAMP
+FILE_TIMESTAMP
 name_mtime (const char *name)
 {
   FILE_TIMESTAMP mtime;
